### ISSUE TITLE
Provide a method in the simulator API that returns the frame rate

### ISSUE THREAD

**Issue by bzinberg** [NextCenturyCorporation/MCS#232]
# TLDR
 In order to reason about physical plausibility, we need the simulator API to tell us the frame rate (number of frames per second in simulated world's time).

# Details
Consider a scene rendered in the simulator with frame rate `r` frames per (simulated) second.  In real scenes unfolding in real time, `r` can be directly perceived; but for sets of frames rendered from a simulator, `r` cannot be observed because the simulated world's time is not the same as the time taken by the simulator running on a computer (e.g., on my laptop, the simulation appears much slower than the GIFs rendered out e.g. [here](https://github.com/NextCenturyCorporation/MCS/blob/development/machine_common_sense/scenes/videos/object_permanence_plausible.gif)).  Below we make a case that plausibility of an observed scene depends on the value of `r`, hence the need to have that information available in the simulator API.

Consider a scene where an object slides on the floor and is subject only to the force of friction.  Such an object will experience a deceleration of magnitude `abs(a) = μ * g`, where `μ` is the coefficient of kinetic friction, which depends on the materials of the object and the floor, and `g` is a physical constant; neither of these quantities depends on `r`.

Because the magnitude of the deceleration is constant until the object comes to a standstill, we can compute it by finite differences over frames 0, 1 and 2.  Assume the object is moving in the _x_ direction, let `x_0`, `x_1` and `x_2` be its displacement in frames 0, 1 and 2, and let `Δt := 1/r` denote the amount of physical time between consecutive frames.  Then
```
a = [(x_2 - x_1) / Δt - (x_1 - x_0) / Δt] / Δt

  = [(x_2 - x_1) - (x_1 - x_0)] / Δt^2

  = r^2 * [(x_2 - x_1) - (x_1 - x_0)]
```

Equating the values of `abs(a)` derived in the above two paragraphs, we have
```
μ * g = r^2 * abs((x_2 - x_1) - (x_1 - x_0))
```
hence
```
μ = r^2 * [ abs((x_2 - x_1) - (x_1 - x_0)) / g ]
```
Note that the quantity in brackets `[]` does not depend on `r`.  Therefore, if we vary the frame rate of the simulator, the correct value of `μ` changes too, proportionally to `r^2`.  When we take `r` close to zero (low frame rate), the scene becomes frictionless, whereas when we take `r` large (high frame rate), the coefficient of kinetic friction between the objects grows without bound.

Next, note that some values of the friction coefficient `μ` are plausible, while some are clearly implausible -- especially (in my intuition, not sure about the developmental literature) extremely high values: objects made out of reasonable materials should be able to slide on each other at least a little.

Thus, depending on frame rate, the implied coefficient of kinetic friction may or may not be commonsensical.

I think what we need in order to fix this issue is to provide a method in the simulator API that returns the frame rate.

**Comment by ThomasSchellenbergNextCentury** [comment]
Hey Ben, thanks for the post. Coincidentally, we already have this task added in our internal JIRA backlog (another good reason to start using GitHub to track these tasks for better visibility!). I'm fairly sure we'll be able to incorporate this information for Eval 4 training. In the meantime, please note that the frame rate is 20 FPS, and we have no intention of changing it anytime soon.

**Comment by ThomasSchellenbergNextCentury** [comment]
Adding TA2 implementation info:

Priority: Eval 4

Components: Python API, Unity Env

Blockers: None

What we really want to do is return the amount of time we’re simulating in between each action. We’re doing that in MCSController.SimulatePhysicsOnce: https://github.com/NextCenturyCorporation/ai2thor/blob/master/unity/Assets/Scripts/MCSController.cs#L550-L552

We can return the “frame rate” to Python in the MetadataWrapper class (add to MCSController.generateMetadataWrapper) just by calculating 1 / (PHYSICS_SIMULATION_STEPS * 0.01)

Relates to: https://github.com/NextCenturyCorporation/MCS/issues/245

TA2 internal JIRA ticket: https://nextcentury.atlassian.net/browse/MCS-512

**Comment by kdrumm** [comment]
Fixed in https://github.com/NextCenturyCorporation/MCS/pull/308
